packages/next-extraction-plugin/README.md

```diff
@@ -11,24 +10,47 @@ yarn add --dev @griffel/next-extraction-plugin
 npm install --save-dev @griffel/next-extraction-plugin
 ```
 
+If you haven't installed yet `@griffel/webpack-loader`, please also install it:
+
+```bash
+yarn add --dev @griffel/webpack-loader
+# or
+npm install --save-dev @griffel/webpack-loader
+```
+
+For more details please check [README of `@griffel/webpack-loader`](../webpack-loader/README.md).
+
 ## Usage
 
 In `next.config.js` file you'll need to add the next-plugin from `@griffel/webpack-extraction-plugin` like so:
+
 ```js
 // next.config.js
-const { withGriffelCSSExtraction } = require('@griffel/webpack-extraction-plugin');
+const { withGriffelCSSExtraction } = require('@griffel/next-extraction-plugin');
 
 module.exports = withGriffelCssExtraction()({
   webpack(config) {
     config.module.rules.push({
-      test: /\.(js|jsx|tsx|ts)$/,
+      test: /\.(js|jsx)$/,
+      exclude: /node_modules/,
+      use: [
+        {
+          loader: '@griffel/webpack-loader',
+        },
+      ],
+    });
+
+    // If your project uses TypeScript
+    config.module.rules.push({
+      test: /\.(ts|tsx)$/,
       exclude: /node_modules/,
       use: [
         {
           loader: '@griffel/webpack-loader',
```

Comment:
Also need `babelOptions` here.

Note that in a typical Next app, `@babel/preset-typescript` needs to be installed manually.

Maybe better to use `next/babel` preset which comes with Next, see https://nextjs.org/docs/advanced-features/customizing-babel-config

Reply:
Good call, added 👍